File: data/readable_format/fortran/file1442.f90
program test
    implicit none
    integer :: p(4) = (/2, 4, 1, 3/)
    integer :: expected_first_pass(4) = (/3, 1, 4, 2/)
    integer :: expected_second_pass(4) = (/1, 2, 3, 4/)
    integer :: i

    ! First transformation
    forall (i = 1:4) p(p(i)) = i
    if (all(p == expected_first_pass)) then
        print *, "First pass test passed."
    else
        print *, "First pass test failed."
        call abort()
    endif

    p = (/2, 4, 1, 3/) ! Reset p for the second transformation

    ! Second transformation
    forall (i = 1:4) p(5 - p(i)) = p(5 - i)
    if (all(p == expected_second_pass)) then
        print *, "Second pass test passed."
    else
        print *, "Second pass test failed."
        call abort()
    endif

end program test

File: data/readable_format/cpp/file1442.cpp
#include <iostream>
#include <cstdlib>
#include <cassert>

void transform_and_test() {
    int p[4] = {2, 4, 1, 3};
    int temp[4]; // Temporary array for assignment

    // First transformation
    for(int i = 0; i < 4; ++i) {
        temp[p[i] - 1] = i + 1;
    }
    for(int i = 0; i < 4; ++i) {
        p[i] = temp[i];
    }

    // Test after first transformation
    int expected_first_pass[4] = {3, 1, 4, 2};
    for(int i = 0; i < 4; ++i) {
        assert(p[i] == expected_first_pass[i]);
    }

    int p2[4] = {2, 4, 1, 3}; // Reset p for the second transformation

    // Second transformation
    for(int i = 0; i < 4; ++i) {
        temp[4 - p2[i]] = p2[4 - i - 1];
    }
    for(int i = 0; i < 4; ++i) {
        p2[i] = temp[i];
    }

    // Test after second transformation
    int expected_second_pass[4] = {1, 2, 3, 4};
    for(int i = 0; i < 4; ++i) {
        assert(p2[i] == expected_second_pass[i]);
    }

    std::cout << "All tests passed successfully." << std::endl;
}

int main() {
    transform_and_test();
    return 0;
}